Cas 8405: logging support for Int64 and uInt64 various MSMetaData/MSSummary mods

dmehring opened the pull request:
Added support for logging uInt64 and Int64.
Fixed bug in MSMetaData::_getSubScanProps() which caused incorrect size of data structure to be computed.

_getSubScanProperties() -> getSubScanProps() (private to public) to allow caller to hold the entire property structure so it doesn't have to call for each subscan. => performance improvement

added progress meter when running getSubScanProps()

added MSMetaData::getNRowMap() to allow caller to hold entire data structure rather than having to call another method for each separate subscan => performance improvement

Added method to allow caller to specify that MSMetaData should always cache subscan properties structure since it can be very expensive to recompute if max cache size has already been exceeded.

Added MSMetaData::getSubScanToIntentsMap() to allow caller to retrieve entire data structure rather than having to call another method for each subscan => performance improvement.

Changed several row determination methods from uInt to uInt64 because VLA is not writing MSes with > 100 mega rows, so it is just a matter of time before table interface limitations are going to result in overflows.

Made use of various performance improvements in MSSummary.

juliantaylor commented:
changing to int64 in the public interface breaks the binary compatibility. While I don't believe this is a valuable goal for casacore, it can still be avoided here as just changing this here will not help. All of casacore (and casa) would need a major overheal to support more than 2**32 rows in tables.

juliantaylor commented on ms/MSOper/MSMetaData.cc:
```diff
@@ -252,7 +254,34 @@ uInt MSMetaData::nRows(CorrelationType cType) {
+    }
+    SHARED_PTR<map<SubScanKey, uInt64> > mymap(
+        new map<SubScanKey, uInt64>()
+    );
```
I guess we don't have a `make_shared` in our abstraction? would save a (probably irrelevant) memory allocation.

dmehring replied:
Not that I'm aware of.

juliantaylor commented on ms/MSOper/MSMetaData.cc:
```diff
@@ -3550,28 +3625,30 @@ std::map<SubScanKey, MSMetaData::SubScanProperties> MSMetaData::_getSubScanPrope
-    while (mIter != mEnd) {
+    static const uInt iSize = sizeof(Int);
+    static const uInt dSize = sizeof(Double);
+	static const uInt structSize = 3*dSize + iSize;
```
wrong indentation

juliantaylor commented on ms/MSOper/MSMetaData.cc:
```diff
@@ -3550,28 +3625,30 @@ std::map<SubScanKey, MSMetaData::SubScanProperties> MSMetaData::_getSubScanPrope
     if (_cacheUpdated(mapSize)) {
-    	_subScanProperties = mysubscans;
+        _subScanProperties = mysubscans;
 	}
```
wrong indentation I think

dmehring commented:
fixed indentation.

dmehring commented:
@juliantaylor  I've reverted the uInt64 changes back to uInt and corrected yet another indentation issue. Hopefully this addresses your remaining concerns.

juliantaylor commented:
thanks
